p2p/p2p.go

```diff
@@ -43,19 +43,25 @@ import (
 )
 
 const ProtocolPrefix = "/substrate/dot/2"
+const ProtocolPrefix2 = "/substrate/dot/3"
 const mdnsPeriod = time.Minute
 
 // Service describes a p2p service, including host and dht
 type Service struct {
-	ctx            context.Context
-	host           core.Host
-	hostAddr       ma.Multiaddr
-	dht            *kaddht.IpfsDHT
-	dhtConfig      kaddht.BootstrapConfig
-	bootstrapNodes []peer.AddrInfo
-	mdns           discovery.Service
-	msgChan        chan<- Message
-	noBootstrap    bool
+	ctx               context.Context
+	host              core.Host
+	hostAddr          ma.Multiaddr
+	dht               *kaddht.IpfsDHT
+	dhtConfig         kaddht.BootstrapConfig
+	bootstrapNodes    []peer.AddrInfo
+	mdns              discovery.Service
+	msgChan           chan<- Message
+	noBootstrap       bool
+	statusMessagesRec map[string]bool
```

Comment:
can probably make these `map[common.Hash]bool` since we are hashing the message to identify it